lib/metadata/bucketclient/backend.js

```diff
@@ -27,75 +35,144 @@ class BucketClientInterface {
             });
             this.client = new bucketclient.RESTClient(bootstrap, log);
         }
+        this.init();
+    }
+
+    init() {
+        // get metadata information
+        const beginPath = '/default/metadataInformation/foo';
+        const log = logger.newRequestLogger();
+        this.client.request('GET', beginPath, log, null, null, (err, data) => {
+            if (!err) {
+                try {
+                    const info = JSON.parse(data);
+                    this.metadataVersion = Number(info.metadataVersion);
+                    this.extendedVersion =
+                        this.metadataVersion > METADATA_DEFAULT_VERSION;
+                } catch (error) {
+                    log.error('error parsing metadata information', { error });
+                }
+            } else if (!err.RouteNotFound) {
+                // retry on InternalError and DBAPINotReady
+                return setTimeout(() => { this.init(); }, 0);
+            }
+            this.gotMetadataInfo = true;
+            return null;
+        });
     }
 
     createBucket(bucketName, bucketMD, log, cb) {
+        if (this.gotMetadataInfo === false) {
+            // not really DBAPINotReady, but cannot make request anyway
```

Comment:
It can be `InternalError` including those http such as `ECONNRESET`, `ECONNREFUSE`, ... or those internal of metadata + the actual `DBAPINotReady`.
We're sending back `DBAPINotReady` or something similar saying that we cannot reach metadata now.